### ISSUE TITLE
No documentation for xpath 

### ISSUE THREAD

**Issue by rvansa** [Hyperfoil/qDup#67]
README speaks about `xpath` command but it does not show much of an usage. From source code it looks like it should be `xml` instead but my attempt to use
```yaml
- xml: /path/to/file.xml>some-element ++ <foobar />
```
Failed with some error about not being able to parse that file (copied locally), though it should contain valid XML.

Trying without spaces results in an error like 'ERROR io.hyperfoil.tools.qdup.Local -   E: bash: foobar: No such file or directory'.

Please document intended usage (I want to modify the file in-place).

**Comment by willr3** [comment]
There was a bug in parsing the `xml` scalar. I updated the README.md and should start to fill out the docs now that someone create the folder for us.
The command effectively modifies it in place but it was faster to copy the file to where qDup was running, modify it there, and copy it back

**Comment by rvansa** [comment]
@willr3 Using `==` for modification (`file>xpath == value`) is weird, the operator is traditionally used for comparison, not for assignment. I'd suggest either plain `=`, `:=` or `<-` though the last option would show some assymetry to `->` which takes variable name at the right side, rather than an expression.

The documentation also shows `file>xpath @key=value` - is it necessary to put the space before `@key`? Conceptually the `@key` should be part of the xpath itself (though I understand that you're probably selecting the node by xpath in library code and then finding out the `key` attribute in yaup code). I'd sync this with node assignment syntax.

**Comment by willr3** [comment]
I agree, `==` for modification feels wrong. I wanted a 2 character sequence to match the `++` and `--` but it might be time to re-evaluate the operators and which ones we want. I would say we have a few operations to consider
* **add xml to a parent node found via xpath** This could be adding a child node or adding an attribute. What should this do if the xpath does not exist? Should it create the path structure to always add the value or only add the value if the path exists?
* **set the value of the xpath target** This could set the value of an attribute or set the content of the xml node
* **delete the xpath**. This would delete the attribute or node identified in the xpath if it exists. If the xpath matches multiple nodes then all of them should be deleted.
* **read the value from the xpath into state** We often need to use values from XML when modifying other configuration. Should this always create an array in the state with size equal to the number of times the xpath matched the document or should it be a scalar value if it only matches once?

What operator sequences ('==', '++','--','->') do you think we should use for those and are there any other operations you think we should consider?

**Comment by rvansa** [comment]
* add: To make it more Java(script)-like you could use `+=`. It should not create the path - IMO it should operate on a node *set* as there might be multiple nodes matching the xpath, so it should modify them all. And if you have empty set, do nothing - maybe with a warning log. Also, it should *not* be used for adding an attribute; attributes are just set or unset.
* delete: Had we used `-=` (for symmetry with `+=`) we would need a node on the right-hand-side of the operator. Maybe this is unnecessary as you need to only select the node - to avoid confusion about any 'decrementation' we could just use `file>xpath del` or `file>xpath delete`
* read: Yes, `file>xpath -> MYVAR` should always create an array. Quick google says that if you want first occurence you should use `(/foo/bar)[0]` (the parentheses are necessary).
